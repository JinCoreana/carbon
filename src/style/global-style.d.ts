import { GlobalStyleComponent } from "styled-components";
import { ThemeObject } from "./themes/base";

declare const GlobalStyle: GlobalStyleComponent<
  Record<string, unknown>,
  ThemeObject
>;

export default GlobalStyle;
