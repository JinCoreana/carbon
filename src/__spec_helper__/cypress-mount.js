import * as React from "react";
import { mount as cypressMount } from "@cypress/react";
import CarbonProvider from "../components/carbon-provider/carbon-provider.component";
import I18nProvider from "../components/i18n-provider/i18n-provider.component";
import enGB from "../locales/en-gb";
import sageTheme from "../style/themes/sage/index";

const CypressMountWithProviders = (children) => {
  return cypressMount(
    <CarbonProvider theme={sageTheme}>
      <I18nProvider locale={enGB}>{children}</I18nProvider>
    </CarbonProvider>
  );
};

export default CypressMountWithProviders;
