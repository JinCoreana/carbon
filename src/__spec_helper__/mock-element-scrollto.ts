const setupScrollToMock = () => {
  // need to mock the `scrollTo` method, which is undefined in JSDOM. As we're not actually testing this behaviour, just make it
  // do nothing.
  HTMLElement.prototype.scrollTo = () => {};
};

export default setupScrollToMock;
