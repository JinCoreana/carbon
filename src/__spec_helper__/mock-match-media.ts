let mocked = false;
let _matches = false;
const removeListener = jest.fn();

export const setupMatchMediaMock = () => {
  if (!global.window) {
    return;
  }
  const noop = () => {};
  Object.defineProperty(global.window, "matchMedia", {
    writable: true,
    value: (query: string) => ({
      matches: _matches,
      media: query,
      onchange: null,
      addListener: noop,
      removeListener,
      dispatchEvent: noop,
    }),
  });
  mocked = true;
};

export const mockMatchMedia = (matches: boolean) => {
  if (!mocked) {
    throw new Error(
      "window.matchMedia has not been mocked. Did you call setupMatchMediaMock()?"
    );
  }
  _matches = matches;
  return { removeListener };
};
