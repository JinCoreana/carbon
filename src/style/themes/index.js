// eslint-disable-next-line import/prefer-default-export
export const THEMES = {
  base: 'base',
  none: 'none',
  classic: 'classic',
  mint: 'mint',
  medium: 'medium',
  large: 'large'
};
