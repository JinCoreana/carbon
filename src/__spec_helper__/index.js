import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { setup } from "./mock-match-media";
import setupResizeObserverMock from "./mock-resize-observer";

require("jest-fetch-mock").enableMocks();

setupResizeObserverMock();
setup();
Enzyme.configure({ adapter: new Adapter() });
