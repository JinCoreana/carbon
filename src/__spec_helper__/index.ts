import { enableFetchMocks } from "jest-fetch-mock";
import { setupMatchMediaMock } from "./mock-match-media";
import setupResizeObserverMock from "./mock-resize-observer";

enableFetchMocks();
setupResizeObserverMock();
setupMatchMediaMock();
