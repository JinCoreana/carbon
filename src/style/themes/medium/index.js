import { aegeanTheme } from '..';

export default (() => {
  console.warn(`"Medium Theme" have been renamed to "Aegean".
All references to that theme should be updated to:
import { aegeanTheme } from "carbon-react/lib/style/themes"`);

  return aegeanTheme;
})();
