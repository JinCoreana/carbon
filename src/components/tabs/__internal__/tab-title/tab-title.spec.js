import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import TabTitle from './tab-title.component';
import StyledTabTitle from './tab-title.style';
import { aegeanTheme, baseTheme } from '../../../../style/themes';
import { assertStyleMatch } from '../../../../__spec_helper__/test-utils';

function render(props) {
  return shallow(
    <TabTitle
      title='Tab Title 1'
      dataTabId='uniqueid1'
      { ...props }
    />
  );
}

function renderStyles(props) {
  return TestRenderer.create(
    <StyledTabTitle
      title='Tab Title 1'
      dataTabId='uniqueid1'
      { ...props }
    />
  );
}

describe('TabTitle', () => {
  let wrapper;
  it('renders as expected', () => {
    wrapper = renderStyles();
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders a title as its child with a text passed as a prop', () => {
    const firstTabTitle = 'Tab Title 1';
    wrapper = render();
    expect(wrapper.children()).toHaveLength(1);
    expect(wrapper.children().text()).toEqual(firstTabTitle);
  });

  it('contains custom className if passed as a prop', () => {
    wrapper = render({ className: 'class' });
    expect(wrapper.find('.class').exists()).toEqual(true);
  });

  describe('attributes', () => {
    wrapper = render();
    it('role equals "tab"', () => {
      expect(wrapper.find("[role='tab']").exists()).toEqual(true);
    });
    it('data-element equals "select-tab"', () => {
      expect(wrapper.find("[data-element='select-tab']").exists()).toEqual(true);
    });
    it('data-tabid equals tabId', () => {
      expect(wrapper.find("[data-tabid='uniqueid1']").exists()).toEqual(true);
    });
  });

  describe('when position prop is set to left', () => {
    it('applies proper styles', () => {
      wrapper = renderStyles({ position: 'left' });
      assertStyleMatch(
        {
          backgroundColor: 'transparent',
          borderRight: `2px solid ${baseTheme.disabled.background}`,
          display: 'block',
          height: 'auto',
          marginLeft: '0px'
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          background: 'transparent',
          borderRightColor: baseTheme.colors.hoveredTabKeyline
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );

      assertStyleMatch(
        {
          marginTop: '0'
        },
        wrapper.toJSON(),
        { modifier: ':first-child' }
      );
    });
  });

  describe('when tab is selected', () => {
    it('has aria-selected attribute set to true', () => {
      wrapper = render({ isTabSelected: true });
      expect(wrapper.find('[aria-selected=true]').exists()).toEqual(true);
    });

    it('applies proper styling', () => {
      wrapper = renderStyles({ isTabSelected: true });
      assertStyleMatch(
        {
          color: baseTheme.text.color,
          backgroundColor: 'transparent',
          borderBottomColor: baseTheme.colors.primary
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          background: 'transparent',
          borderBottomColor: baseTheme.colors.primary,
          color: baseTheme.text.color
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );
    });

    describe('when position prop is set to left', () => {
      it('applies proper styling ', () => {
        wrapper = renderStyles({ position: 'left', isTabSelected: true });

        assertStyleMatch(
          {
            borderRightColor: baseTheme.colors.primary,
            backgroundColor: 'transparent'
          },
          wrapper.toJSON()
        );

        assertStyleMatch(
          {
            background: 'transparent',
            borderRightColor: baseTheme.colors.primary
          },
          wrapper.toJSON(),
          { modifier: ':hover' }
        );
      });
    });
  });

  describe('when tab is not selected', () => {
    it('has aria-selected attribute set to false', () => {
      wrapper = render({ isTabSelected: false });
      expect(wrapper.find('[aria-selected=false]').exists()).toEqual(true);
    });
  });

  describe('when tab has warning', () => {
    it('applies proper styling', () => {
      wrapper = renderStyles({ tabHasWarning: true });
      assertStyleMatch(
        {
          borderBottomColor: baseTheme.colors.warning
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          borderBottomColor: baseTheme.colors.warning
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );
    });

    it('applies proper styling when position is "left"', () => {
      wrapper = renderStyles({ position: 'left', tabHasWarning: true });
      assertStyleMatch(
        {
          borderRightColor: baseTheme.colors.warning
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          borderRightColor: baseTheme.colors.warning
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );
    });
  });

  describe('when tab has error', () => {
    it('applies proper styling', () => {
      wrapper = renderStyles({ tabHasError: true });
      assertStyleMatch(
        {
          borderBottomColor: baseTheme.colors.error
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          borderBottomColor: baseTheme.colors.error
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );
    });

    it('applies proper styling when position is "left"', () => {
      wrapper = renderStyles({ position: 'left', tabHasError: true });
      assertStyleMatch(
        {
          borderRightColor: baseTheme.colors.error
        },
        wrapper.toJSON()
      );

      assertStyleMatch(
        {
          borderRightColor: baseTheme.colors.error
        },
        wrapper.toJSON(),
        { modifier: ':hover' }
      );
    });
  });

  describe('when in DLS theme', () => {
    it('applies proper paddings', () => {
      wrapper = renderStyles({ theme: aegeanTheme });
      assertStyleMatch(
        {
          padding: '14px 16px 12px'
        },
        wrapper.toJSON()
      );
    });

    it('applies proper outline style on focus', () => {
      wrapper = mount(
        <ThemeProvider theme={ aegeanTheme }>
          <TabTitle
            title='Tab Title 1'
            dataTabId='uniqueid1'
            theme={ aegeanTheme }
            position='top'
            isTabSelected
          />
        </ThemeProvider>
      ).find(TabTitle);

      wrapper.simulate('focus');
      assertStyleMatch({ position: 'relative' }, wrapper, { modifier: ':focus' });
      assertStyleMatch({
        position: 'absolute',
        boxShadow: 'inset 2px 0 0 0 #FFB500, inset -2px 0 0 0 #FFB500, inset 0 2px 0 0 #FFB500, 0 2px 0 0 #FFB500'
      }, wrapper, { modifier: ':focus:after' });
    });

    describe('and position set to left', () => {
      it('applies proper outline style on focus', () => {
        wrapper = mount(
          <ThemeProvider theme={ aegeanTheme }>
            <TabTitle
              title='Tab Title 1'
              dataTabId='uniqueid1'
              theme={ aegeanTheme }
              position='left'
              isTabSelected
            />
          </ThemeProvider>
        ).find(TabTitle);

        wrapper.simulate('focus');
        assertStyleMatch({ position: 'relative' }, wrapper, { modifier: ':focus' });
        assertStyleMatch({
          position: 'absolute',
          boxShadow: 'inset 2px 0 0 0 #FFB500, 2px 0 0 0 #FFB500, inset 0 2px 0 0 #FFB500, inset 0 -2px 0 0 #FFB500'
        }, wrapper, { modifier: ':focus:after' });
      });
    });
  });
});
