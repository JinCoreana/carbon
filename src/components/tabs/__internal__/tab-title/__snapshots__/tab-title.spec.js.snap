// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabTitle renders as expected 1`] = `
.c0 {
  background-color: transparent;
  border-bottom: 2px solid #E6EBED;
  color: rgba(0,0,0,0.55);
  display: inline-block;
  font-weight: bold;
  height: 100%;
  line-height: 20px;
  margin: 0;
  padding: 14px 16px 12px;
}

.c0:first-child {
  margin-left: 0;
}

.c0:hover,
.c0:focus {
  background: transparent;
  border-bottom-color: #4DB84D;
  color: rgba(0,0,0,0.9);
  outline: none;
}

<li
  className="c0"
  title="Tab Title 1"
/>
`;
